### PR TITLE
fix: Memory leakage case of dummy segment

### PR DESCRIPTION
**Pull request by sawadashota**
*Description of changes:*

#327 causes memory leakage when dummy segment. So this PR fix it.

### CODE REVIEW

**Review by nikolaianohyn — APPROVED** [review]

**Review by mathn00b — APPROVED** [review]

**Review by prigornitskiy — APPROVED** [review]

File: xray/segment.go [hunk 1]
```diff
@@ -357,21 +357,22 @@ func (seg *Segment) Close(err error) {
 		seg.addError(err)
 	}
 
+	cancelSegCtx := seg.cancelCtx
```
Comment (nikolaianohyn) [inline xray/segment.go:360]: another way: don't create routine for Sampled segments

```
	// create a new context to close it on segment close;
	// this makes sure the closing of the segment won't affect the client's code, that uses the returned context
	if !seg.Dummy {
		ctx1, cancelCtx := context.WithCancel(ctx)
		seg.cancelCtx = cancelCtx
		go func() {
			<-ctx1.Done()
			seg.handleContextDone()
		}()
	}
```

Comment (nikolaianohyn) [reply xray/segment.go:360]: For me it's better because you don't send sampled segments and don't create extra go routines

**Review by NathanielRN — COMMENTED** [review]
Thanks for this change! I had some questions but this is looking good!

File: xray/segment.go (hunk 1 above)
Comment (nikolaianohyn) [reply xray/segment.go:360]: Hi @NathanielRN [here PR](https://github.com/aws/aws-xray-sdk-go/pull/365)

Comment (sawadashota) [reply xray/segment.go:360]: Thanks @nikolaianohyn ! I think your PR is better so I will close this PR.

**Pull request closed without merging**